Core/Src/main.c

```diff
@@ -87,14 +92,40 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+
+  static const uint8_t START = 1;
+  static const uint8_t END = 2;
+  static const uint16_t MAX_ADC = 1023; // 1023 is the max value that can be stored by a 10 bit value
+
+  uint16_t adc_value = 0x0000;
+  uint16_t compare_value;
+  static const uint16_t SCALE_VALUE = 3000; // Scaling the 60000 to the period of 20ms: 60000/20=3000
+
+  HAL_GPIO_WritePin(GPIOA, GPIO_PIN_5, GPIO_PIN_SET);
+
   while (1)
   {
+	  HAL_GPIO_WritePin(GPIOA, GPIO_PIN_5, GPIO_PIN_RESET);
+
+	  HAL_SPI_TransmitReceive(&hspi1, spi_tx_buffer, spi_rx_buffer, sizeof(spi_tx_buffer), 1000);
+
+	  HAL_GPIO_WritePin(GPIOA, GPIO_PIN_5, GPIO_PIN_SET);
+
+	  adc_value = ((uint16_t) spi_rx_buffer[1] << 8) | (uint16_t) spi_rx_buffer[2];
+
+	  compare_value = ((double)adc_value/MAX_ADC)*(END-START)*SCALE_VALUE+START;
```

Comment:
Be careful here, I think that you need to multiply the entire expression by SCALE_VALUE, I think that the START is not currently being scaled.
Also more descriptive naming might make it clearer when reading (Eg. MIN_PWM_MS, or MIN_PWM_COUNTS) depending which way you go.